sdks/java/maven-archetypes/examples/pom.xml

```diff
@@ -43,9 +43,37 @@
       </extension>
     </extensions>
 
+
+    <resources>
+      <resource>
```

Comment:
Can you add comments (and teach me too!) how this works?

My best understanding so far:
* The first resource statement says to take the `archetype-resources/pom.xml`, enable filtering (replacements) on it, and place it into `src/main/resources` in the target JAR. (This is actually a two-phase process; the middle step being in the local `target/` directory.)
* The second resource statement says to skip copying the same file without filtering to avoid possible collisions.
* Unclear how this interacts with the behavior of the archetype plugin.